.github/workflows/build-test.yml

```diff
@@ -0,0 +1,31 @@
+name: Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
```

Comment:
Note that this does need to be `pull_request_target` if any of our action workflows require secrets, since those won't be provided on forks :thinking: 

`pull_request` seems perfectly fine to use on repos _without_ secrets though - maybe even preferable, as testing changes to the workflows is easier with `pull_request`. So maybe this is fine as a default.

Reply:
Since [we want to add npm publishing](https://github.com/MetaMask/metamask-module-template/issues/3) eventually, this seems like an important suggestion, though I'm not sure it needs to block this PR.

Reply:
Publishing would run post-merge, so that would be from the `push` event. I don't think using `pull_request` poses any challenges there.

My comment was more about status checks that use secrets, such as on `metamask_extension` where our CI builds have valid non-production Sentry and Matomo keys.